server/libs/ffbinaries/index.js

```diff
@@ -1,315 +0,0 @@
-const os = require('os')
-const path = require('path')
-const axios = require('axios')
-const fse = require('../fsExtra')
-const async = require('../async')
-const StreamZip = require('../nodeStreamZip')
-const { finished } = require('stream/promises')
-
-var API_URL = 'https://ffbinaries.com/api/v1'
```

Comment:
@mikiher @advplyr Switching from `https://ffbinaries.com/api/v1` to `https://api.github.com/repos/${this.owner}/${this.repo}/releases/tags/${releaseTag}` is a bad idea because unauthenticated GitHub's API calls are severely rate-limited and can cause the server to fail launching. See this issue: https://github.com/advplyr/audiobookshelf/issues/3266

Reply:
The rate limit is pretty strict.
https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28

Probably best to revert back to ffbinaries or we could host the binaries ourselves.

Docker builds should also not be installing ffmpeg/ffprobe since they are packaged in the image. Only reason they are is because the image is installing 6.1 but the BinaryManager is expecting 5.1.

I updated the BinaryManager to 5.1 because of an error with ffprobe on 6.1 that I was able to reproduce on Windows https://github.com/advplyr/audiobookshelf/issues/2689

Reply:
IMO for Docker the dependencies should always be bundled inside. This is IMO one of the main points of using a Docker image: make running containers more predictable.

Reply:
Yeah I agree. I'm testing ffmpeg 5.1 in the Dockerfile now

Reply:
The ffprobe 6.1 issue with those mp3 files mentioned in #2689 is specific to Windows in my testing. Since it is not easy to include an older version of a package with alpine linux I think I'll just add a skip for the BinaryManager for docker installs for now.

Reply:
I didn't anticipate rate limits to be an issue, since I meant for the download happen only _once_ per server.

However @advplyr already commented above that:
> I realize that the docker images are going to download the binaries on every image start since they aren't getting mapped outside the container. Also the Dockerfile is already including ffmpeg/ffprobe but the latest version.

And now because of recent failures, some docker containers have be repeatedly going up and down.

I still think if we move the default download target to a stable location like configPath, like @advplyr suggested above, we should be safe, but I also agree that the rate limits are pretty severe and might be quickly reached in some circumstances. I'll revert to ffbinaries.

> IMO for Docker the dependencies should always be bundled inside. This is IMO one of the main points of using a Docker image: make running containers more predictable.

I just wanted to note, though, that even in the current docker setup, binaries aren't really bundled, but installed through apk (which probably doesn't have the rate-limiting issue, but is still downloaded over the network).

Reply:
> I just wanted to note, though, that even in the current docker setup, binaries aren't really bundled, but installed through apk (which probably doesn't have the rate-limiting issue, but is still downloaded over the network).

This only happens when the image is built though. Once it is built the users system doesn't run it.

Docker build can stay with ffmpeg/ffprobe v6.1. It is the Windows built that had the issue with 6.1.
It would be nice to have them all use the same version but it is not easy to install older alpine linux packages that I could see so I was thinking of just having the binary manager skipped for docker.

Reply:
Sorry for speaking out of ignorance.
Then I conceded that BinaryManager is not suitable for docker installations (at least not as part of server init).

Reply:
I set it up to skip the binary manager if the source is "docker" that is set in index.js. That should do it

Reply:
@mikiher I apologize for the tone of my previous comment, I reworded it.

Reply:
No worries, I'm using untone() so I'm tone-insensitive...😁